docs/securityfaq.md

```diff
@@ -0,0 +1,27 @@
+# SMART Health Links Security FAQ (draft)
+
+## Can someone guess my SHL and access my data?
+
+A SHL manifest URL contains at least 256 bits of entropy, which makes guessing a valid URL unfeasible.
+
+## Can someone modify my SHL?
+
+The SHL manifest is encrypted using AES-CGM, which provides integrity as well as confidentiality; guaranteeing that only an entity knowing the encryption key could have produced the content. For individual files contained in the manifest, SHCs are signed by a specific issuers and therefore provide strong integrity guarantees; SMART API access and directly encoded FHIR resources rely on the integrity of the retrieval access point. 
+
+## Can someone with my SHL access my data?
+
+Someone with access to the SHL (or able to take a picture of its QR encoding) can retrieve the data. If the data is sensitive, a Passcode should be used.
+
+## What are the risks of clicking or scanning a SHL?
+
+Applications and libraries should implement conventional input sanitization to make sure encoded SHL, SHL URL, and QR-encoded SHL are well-formed before interpreting the data.
+
+## Can I be phished using a SHL?
+
+Like any other web-based access data, a SHL could be used to mount a phishing attack against a user, e.g., to learn login information to an EHR or the Passcode of a SHL. Conventional anti-phishing measures should be followed by applications and libraries to minimize this risk. 
+
+## Can someone invalidate my SHL by guessing my Passcode?
+If a Passcode is used, then enough bad guesses will invalidate the link. A malicious party could perform such a denial of service attack if it has access to the SHL URL; care must be taken to minimize exposure of such link. As a fallback, a user can obtain a fresh SHL. Conventional denial of service countermeasures can be employed to filter repeated attacks.
+
+## Can someone snoop the retrieval of a SHL and manifest?
```

Comment:
Should we explicitly mention "passcode phishing" (e.g., attacker tricks a user into revealing a passcode, perhaps by setting up a fake SHL to emulate a real one)?

Reply:
It's mentioned in the phishing question. Do you think I should add more details?

Reply:
Oh, I think that covers the ground well!

Maybe just worth explaining who the "I" there (like, a data recipient can be phished to reveal information from the data sharer's record, which is a challenging kind of split responsibility).

Reply:
I must say I was a bit confused on how to interpret the two roles defined in the spec (data sharer, data recipient), with the three roles I have in my mental model: issuer (creating the SHL), user (person to whom the data pertains), verifier (consumer of the data). For processing purposes, you have a sharer and a recipient (with potential parties (users) simply forwarding the link or carrying it in the form of a QR code), but for security analysis it is useful to think about these roles separately (e.g., there might be phishing risks when a user retrieves SHL from an EHR system _and_ when they try to present it).

Reply:
Here's a suggestion (happy to adjust based on feedback!): https://github.com/smart-on-fhir/smart-health-links/pull/8